src/System.Security.Cryptography.X509Certificates/tests/CertificateCreation/CertificateRequestChainTests.cs

```diff
@@ -10,27 +10,38 @@ namespace System.Security.Cryptography.X509Certificates.Tests.CertificateCreatio
 {
     public static class CertificateRequestChainTests
     {
+        const string RootDN = "CN=Experimental Root Certificate";
+        const string Intermed1DN = "CN=First Intermediate Certificate, O=Experimental";
+        const string Intermed2DN = "CN=Second Intermediate Certificate, O=Experimental";
+        const string LeafDN = "CN=End-Entity Certificate, O=Experimental";
+
         public static bool PlatformSupportsPss { get; } = DetectPssSupport();
 
-        [Fact]
-        public static void CreateChain_ECC()
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public static void CreateChain_ECC(bool useExtraStore)
```

Comment:
The tests in this file are really just making sure that the CertificateRequest class is building certificates such that they properly chain.

The tests for using newly created chains to test functionality of X509Chain should really be in DynamicChainTests.cs .. it helps with scoping to see what component is failing.  (And once in DynamicChainTests.cs the tests can use `TestDataGenerator.MakeTestChain3` to make a 3-deep chain appear out of nowhere, or new helpers as warranted).

I think ideally the changes in this file would all be reverted, as it then maximizes how compatible this file is with .NET Framework.